create-image.sh

```diff
@@ -1,11 +1,36 @@
 #!/usr/bin/env bash
 
-source image.conf
-if [ $? -ne 0 ]; then
-    echo "ERROR: could not get configuration from image.conf"
-    exit 1
+CONFIG="image.conf"
+
+show_usage() {
+    echo "Create Arch Linux image for the Risc-V StarFive VisionFive 2 single board computer\n
+
+USAGE:
+    create-image.sh [ -h | CONFIG_FILE]
+
+ARGS:
+    <FILE>    Configuration file
+
+OPTIONS:
+    -h, --help        Print this help message."
+}
+
+
+if [ $# -ne 0 ]; then
+    [ $# -gt 1 ] && (>&2 echo "error: too many arguments" && show_usage && exit 1)
```

Comment:
Is there any particular reason why we are exiting just because the user provided extra argument(s)? I don't think that we are accepting any arguments other than the config file. So we can just ignore rest of the arguments.

Reply:
Extra arguments mean that the user didn't use the command as expected. We should error on that. Silently ignoring sets the wrong expectation from the user and the behaviour is effectively undefined.

Reply:
>  Silently ignoring sets the wrong expectation from the user and the behaviour is effectively undefined.

But we aren't even using `$2`, why would that matter? I don't mean to be rude but I legitimately do not understand.

The presence or absence of any additional arguments don't affect the control flow of this script, so imho, it shouldn't matter.

Reply:
> But we aren't even using `$2`, why would that matter? I don't mean to be rude but I legitimately do not understand.
> 
> The presence or absence of any additional arguments don't affect the control flow of this script, so imho, it shouldn't matter.

The presence of unexpected input shouldn't be ignored, it should error. We are validating that the input provided is correct. It's just a best practice.